core/project_settings.cpp

```diff
@@ -647,8 +644,7 @@ Error ProjectSettings::_save_settings_binary(const String &p_file, const Map<Str
 	FileAccess *file = FileAccess::open(p_file, FileAccess::WRITE, &err);
 	if (err != OK) {
 
-		ERR_EXPLAIN("Couldn't save project.binary at " + p_file);
-		ERR_FAIL_COND_V(err, err);
+		ERR_FAIL_COND_V_MSG(err, err, "Couldn't save project.binary at " + p_file + ".");
```

Comment:
The `if` can be removed, and I'd make the condition more explicit: `ERR_FAIL_COND_V_MSG(err != OK, err, ...)`.